torch/csrc/jit/codegen/cuda/lower_utils.cpp

```diff
@@ -206,6 +206,48 @@ kir::Expr* applyReplacements(
   }
 }
 
+c10::optional<IterDomain*> getMaybeWarpReductionDim(const ReductionOp* node) {
+  auto fuser_tv_out = node->out()->as<TensorView>();
+  auto fuser_tv_in = node->in()->as<TensorView>();
+
+  // only support reducing to registers for now.
+  if (fuser_tv_in->getMemoryType() != MemoryType::Local ||
+      fuser_tv_out->getMemoryType() != MemoryType::Local) {
+    return c10::nullopt;
+  }
+
+  IterDomain* reduction_on_xdim = nullptr;
+  for (auto id : fuser_tv_out->domain()->domain()) {
+    if (id->isReduction()) {
+      if (id->getParallelType() == ParallelType::TIDx) {
+        reduction_on_xdim = id;
+      } else {
+        return c10::nullopt;
+      }
+    }
```

Comment:
Is it fine to have non-parallelized reduction axes too? I think yes.

Reply:
Yes I think serial reductions should work too. Originally excluded them for simplicity. I will add test.